### ISSUE TITLE
Protractor typescript 2.4.1 

### ISSUE THREAD

**Issue by reigelj** [mgechev/angular-seed#2013]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see use [gitter](https://gitter.im/mgechev/angular2-seed) or [stackoverflow](https://stackoverflow.com/questions/tagged/angular2)
```

**Current behavior**
After the recent typescript 2.4.1 update, build.js.e2e now returns:
```
[09:33:50] Starting 'build.js.e2e'...
src\e2e\specs\about.component.e2e-spec.ts(10,20): error TS2559: Type 'By' has no properties in common with type 'Locator'.
src\e2e\specs\app.component.e2e-spec.ts(10,40): error TS2345: Argument of type '"Welcome to angular-seed!"' is not assignable to parameter of type 'Expected<Promise<string>>'.
src\e2e\specs\app.component.e2e-spec.ts(14,20): error TS2559: Type 'By' has no properties in common with type 'Locator'.
src\e2e\specs\app.component.e2e-spec.ts(18,20): error TS2559: Type 'By' has no properties in common with type 'Locator'.
src\e2e\specs\app.component.e2e-spec.ts(22,20): error TS2559: Type 'By' has no properties in common with type 'Locator'.
src\e2e\specs\home.component.e2e-spec.ts(10,20): error TS2559: Type 'By' has no properties in common with type 'Locator'.
src\e2e\specs\home.component.e2e-spec.ts(14,20): error TS2559: Type 'By' has no properties in common with type 'Locator'.
src\e2e\specs\home.component.e2e-spec.ts(19,13): error TS2559: Type 'By' has no properties in common with type 'Locator'.
src\e2e\specs\home.component.e2e-spec.ts(20,13): error TS2559: Type 'By' has no properties in common with type 'Locator'.
src\e2e\specs\home.component.e2e-spec.ts(22,20): error TS2559: Type 'By' has no properties in common with type 'Locator'.
```

**Expected behavior**
Ideally, build.js.e2e should finish without any errors, allowing the user to run protractor. 
The ideal behavior is reached after modifying package.json devDependencies :
"@compodoc/compodoc": "1.0.0-beta.10",
"typescript": "~2.3.4" ,

Note: This is a nice way to ignore the strict 2.4 typescript for now. However, it means that typescript will be stuck in 2.3.4 until protractor's issue is fixed. 

Suspected cause: Protractor's issue. Typescript 2.4 introduced weak type checking.  Protractor 4.x has a weak typing issue. Upgrading to protractor 5.x gives the e2e build error: node_modules/protractor/built/browser.d.ts(16,18): error TS2430: Interface 'AbstractExtendedWebDriver' incorrectly extends interface 'ExtendedWebDriver'.

**Minimal reproduction of the problem with instructions**
1. Clone angular-seed-master
2. Install npm modules (this will install the newest version of typescript)
3.  npm run build.e2e (or another command that calls the gulptask build.e2e.js)

**What is the motivation / use case for changing the behavior?**
The motivation for changing this behavior is to allow users, while waiting for protractor to update the interface extension issue, to build e2e tests without having to manually roll back typescript/compodoc.

**Please tell us about your environment:**
Windows 10, 64 bit
npm- 5.0.4

* **Angular Seed Version:** fb2549a

* **Node:** `node --version` = v6.9.1

**Comment by mgechev** [comment]
Unfortunately, this is issue in the protractor's type definitions, not Angular seed.

**Comment by RomkeVdMeulen** [comment]
Fortunately this was [fixed in Protractor 5.1.0](https://github.com/angular/protractor/blob/5.1.0/lib/locators.ts#L19).
